lib/Transforms/MemoryBanking.cpp

```diff
@@ -125,6 +171,76 @@ SmallVector<Value, 4> createBanks(Value originalMem, uint64_t bankingFactor,
             banks.push_back(bankAllocaOp);
           }
         })
+        .Case<memref::GetGlobalOp>([&](memref::GetGlobalOp getGlobalOp) {
```

Comment:
How difficult would it be to place all this code in a separate function and just call it here? It is quite large, but unsure if that would require too many arguments.